### ISSUE TITLE
Inconsistency between the reported loss and the actual loss used for gradient computation

### ISSUE THREAD

**Issue by pzelasko** [k2-fsa/snowfall#194]
The loss we backprop is normalized by the number of supervisions:

https://github.com/k2-fsa/snowfall/blob/5d1b00dd2ab4c809714f588fa6d2487cde5ea46c/egs/librispeech/asr/simple_v1/mmi_att_transformer_train.py#L114-L117

But the loss we report is normalized by the number of frames:

https://github.com/k2-fsa/snowfall/blob/5d1b00dd2ab4c809714f588fa6d2487cde5ea46c/egs/librispeech/asr/simple_v1/mmi_att_transformer_train.py#L267-L278

It looks like it wasn't intended. I think the latter makes more sense to me to use in backprop (but we'd probably need to re-tune learning rates etc.) - WDYT?

**Comment by danpovey** [comment]
Yes, I think normalizing by number of frames would be OK.  With Adam optimizers this shouldn't make a difference to the results. 
My preference in the abstract would be to simply not normalize at all, which would mean we wouldn't have to take into account accum_grad.  But I think it's traditional in machine learning to normalize somehow, so IDK whether people feel this might be confusing to readers.

**Comment by pzelasko** [comment]
Could also be surprising if we want to try out a different optimizer that doesn't have Adam-like gradient scaling.
